config_defaults_inc.php

```diff
@@ -541,6 +541,20 @@
 $g_validate_email = ON;
 
 /**
+ * Enable support for logging in by email and password, in addition to
+ * username and password.  This will only work as long as there is a single
+ * user with the specified email address and the email address is not blank.
+ *
+ * @global integer $g_email_login_enabled
+ */
+$g_email_login_enabled = ON;
+
```

Comment:
Setting to ON means having two user names matching a password.
This is decreased security.

There are users that use the same password but different user names for various accounts.
They shouldn't, but they do.
Typically their email address is known to others but not their user name.
Offering email address as user name is decreased security in this case.

I am not telling that we are opening a serious security issue with this, but keep in mind that hackers are using each single information they have.

> added a check that the provided email is a valid email address before looking up the DB for emails.

Not sure if you are aware that we still have the additional load with each SOAP call.